File: src/arboles/java/Arboles.java
package arboles.java;

import java.util.Scanner;

public class Arboles {
    private static final String CLEAR = "\033[H\033[2J";

    public static void main(String[] args) {
        Tree miTree = new Tree();
        short option;
        int data;
        Scanner sc = new Scanner(System.in);

        do {
            System.out.print(CLEAR);

            System.out.println("Menu:");
            System.out.println("1. Insert.");
            System.out.println("2. Remove.");
            System.out.println("3. Find.");
            System.out.println("4. Show pre-order.");
            System.out.println("5. Show in-order.");
            System.out.println("6. Show post-order.");
            System.out.println("7. Show level-order.");
            System.out.println("8. Exit.");
            System.out.println("Choose one:.");
            option = sc.nextShort();

            System.out.print(CLEAR);

            switch (option) {
                case 1 -> {
                    System.out.println("\nInsert a new element:");
                    data = sc.nextInt();
                    miTree.add(data);
                }
                case 2 -> {
                    System.out.println("\nInsert the element to remove:");
                    data = sc.nextInt();
                    miTree.delete(data);
                }
                case 3 -> {
                    System.out.println("\nInsert the element to search:");
                    data = sc.nextInt();
                    System.out.println(miTree.containsNode(data));
                }
                case 4 -> miTree.traversePreOrder(miTree.getRoot());
                case 5 -> miTree.traverseInOrder(miTree.getRoot());
                case 6 -> miTree.traversePostOrder(miTree.getRoot());
                case 7 -> miTree.traverseLeverOrder();
                case 8 -> System.out.println("Bye...");
                default -> System.out.println("\\nWrong data, please try again.");
            }
            System.out.println("\n\nPress ENTER to continue...");
            sc.next();
        } while (option != 8);
    }
}

File: src/arboles/kotlin/Arboles.kt
package arboles.kotlin

fun main() {
    val miTree = Tree()
    var option: Int
    var data: Int

    do {
        println("Menu:")
        println("1. Insert.")
        println("2. Remove.")
        println("3. Find.")
        println("4. Show pre-order.")
        println("5. Show in-order.")
        println("6. Show post-order.")
        println("7. Show level-order.")
        println("8. Exit.")
        println("Choose one:.")
        option = readLine()!!.toInt()

        when (option) {
            1 -> {
                println("\nInsert a new element:")
                data = readLine()!!.toInt()
                miTree.add(data)
            }
            2 -> {
                println("\nInsert the element to remove:")
                data = readLine()!!.toInt()
                miTree.delete(data)
            }
            3 -> {
                println("\nInsert a element to search:")
                data = readLine()!!.toInt()
                miTree.containsNode(data)
            }
            4 -> miTree.traversePreOrder(miTree.getRoot())
            5 -> miTree.traverseInOrder(miTree.getRoot())
            6 -> miTree.traversePostOrder(miTree.getRoot())
            7 -> miTree.traverseLevelOrder()
            8 -> println("Bye...")
            else -> println("\\nWrong data, please try again.")
        }

        println("\n\nPress ENTER to continue...")
        readLine()
    } while (option != 8)
}